### PR TITLE
Feature: APP-2383 - Prepare App for Base Network 

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description

- adds support for Base and Base Goerli Networks (disabled in DAO creation until new SDK comes in)
- initial refactor of the providers

Task: [APP-2383](https://aragonassociation.atlassian.net/browse/APP-2383)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [x] I have tested my code on the test network.

[APP-2356]: https://aragonassociation.atlassian.net/browse/APP-2356?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[APP-2383]: https://aragonassociation.atlassian.net/browse/APP-2383?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/context/providers.tsx [hunk 1]
```diff
@@ -1,139 +1,236 @@
...
-    });
-  } else {
-    return new InfuraProvider(CHAIN_METADATA[network].id, infuraApiKey);
+/* CONTEXT CONSUMER ========================================================= */
+
+export function useProviders(): NonNullable<Providers> {
+  const context = useContext(ProvidersContext);
+
+  // Check if context is defined
+  if (context === undefined) {
```
Comment (cgero-eth) [inline src/context/providers.tsx]: We should probably check for `null` values as well
```suggestion
  if (context == null) {
```

File: src/context/providers.tsx [hunk 2]
```diff
@@ -1,139 +1,236 @@
...
+export function useSpecificProvider(
+  chainIdOrNetwork: SupportedChainID | SupportedNetworks
+): AlchemyProvider | InfuraProvider | JsonRpcProvider | null {
+  const [provider, setProvider] = useState<
+    AlchemyProvider | InfuraProvider | JsonRpcProvider | null
+  >(null);
+
+  useEffect(() => {
+    setProvider(getApiProvider(chainIdOrNetwork));
+  }, [chainIdOrNetwork]);
```
Comment (cgero-eth) [inline src/context/providers.tsx]: Do we need to store the provider as state here or can we just use `useMemo` like:
```suggestion
  const provider = useMemo(
    () => getApiProvider(chainIdOrNetwork),
    [chainIdOrNetwork]
  );

  return provider;
```

**Review by cgero-eth — COMMENTED** [review]
Nice cleanup!

File: src/context/providers.tsx (hunk 2 above)
Comment (Fabricevladimir) [reply src/context/providers.tsx]: Fair point, considering there are no actual side effects and DOM manipulation, we can just return the memoized value

**Comment by okhaimie-dev** [comment]
Yes! I am excited.

I was literarily thinking about doing this today, but I have been struggling with deploying the contract to base blockchain. Glad this is getting released soon.

**Review by cgero-eth — APPROVED** [review]
Thanks @Fabricevladimir , well done!

**Pull request merged**